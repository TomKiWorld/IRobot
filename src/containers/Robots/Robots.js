import React from 'react';

import SearchBox from '../../components/SearchBox/SearchBox';
import Scroll from '../../components/Scroll/Scroll';
import ErrorBoundry from '../../components/ErrorBoundry/ErrorBoundry';
import CardList from '../../components/CardList/CardList';

const Robots = ({ isPending, onSearchChange, filteredRobots }) => {
  return (
    <main className='container'>
      <SearchBox searchChange={onSearchChange}/>
        <Scroll>
        { isPending ? <h1>Loading</h1> :
          <ErrorBoundry>
            <CardList robots={filteredRobots} />
          </ErrorBoundry> 
        }
        </Scroll>
    </main>
  );
};

export default Robots;
