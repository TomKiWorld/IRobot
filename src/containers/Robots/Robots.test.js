import { shallow } from 'enzyme';
import React from 'react';
import Robots from './Robots';

describe('Robots Component', () => {
  it('Expect to render Robots Component pending false', () => {
    expect(shallow(<Robots isPending={false}/>)).toMatchSnapshot();
  });

  it('Expect to render Robots Component pending true', () => {
    expect(shallow(<Robots isPending={true}/>)).toMatchSnapshot();
  });
});
