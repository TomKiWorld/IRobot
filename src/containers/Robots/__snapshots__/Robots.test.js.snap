// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Robots Component Expect to render Robots Component pending false 1`] = `ShallowWrapper {}`;

exports[`Robots Component Expect to render Robots Component pending true 1`] = `ShallowWrapper {}`;
